### DISCUSSION TITLE
Do Updated Lifestyle Hooks support Nested Collections?

### DISCUSSION THREAD

**Discussion by Progression** [livewire/livewire#3234] [1. Help] [answered]
I have an Eloquent collection that I am binding too - from my Livewire component:

```
public $actionPoints;

protected $rules = [
        'actionPoints.*.action_point' => 'string'
];
```

And I want to be able to catch these specific properties when they are updated but I am struggling see if there is a specific syntax that works for 'updated' functions which correspond to properties that contain wildcards? The following works for specific items in the collection:

```
public function updatedActionPoints0ActionPoint($value) {
        ...
 }
```

Though obviously not very practical if you have an unknown number of items in the collection.

And I know I will probably have to make do with the "updated" function and filter over the name but just thought I'd see if there was an alternative. I did check the Livewire source and can't see anything relevant to wildcards but I'm back programming after a long hiatus so my php/laravel/coding general skills are pretty rusty!

**Comment by joshhanley** [comment, marked as answer]
@Progression you can actually use `updatedActionsPoints` hook and make use of the second parameter (not documented), which is `$key` to work out exactly which record and property has been updated. The key should be in dot notation so you can just use Laravel's string helpers to work it out.

So for example

```php
public function updatedActionPoints($value, $key) {
    // Do something here by using $key to work out record
}
```

Hope this helps!

**Reply by Progression** [reply]
Hurrah! That does the job nicely and simplifies things just enough. I knew there had to be something in there somewhere! Cheers

**Reply by joshhanley** [reply]
No worries 🙂
